hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/DeploymentManagementTest.java

```diff
@@ -545,6 +534,44 @@ public void assignDistributionSetAndAutoCloseActiveActions() {
         }
     }
 
+    @Test
+    @Description("Verifies that if multi assignment is enabled previous Distribution is not canceled when a new one is assigned.")
+    @ExpectEvents({ @Expect(type = TargetCreatedEvent.class, count = 10),
+            @Expect(type = TargetUpdatedEvent.class, count = 20), @Expect(type = ActionCreatedEvent.class, count = 20),
+            @Expect(type = DistributionSetCreatedEvent.class, count = 2),
+            @Expect(type = SoftwareModuleCreatedEvent.class, count = 6),
+            @Expect(type = MultiActionEvent.class, count = 2),
+            @Expect(type = TargetAssignDistributionSetEvent.class, count = 0) })
+    public void previousAssignmentsAreNotCanceledInMultiAssignMode() {
+        setMultiAssignmentsEnabled(true);
+        final List<Target> targets = testdataFactory.createTargets(10);
+
+        // First assignment
+        final DistributionSet ds1 = testdataFactory.createDistributionSet("Multi-assign-1");
+        assignDistributionSet(ds1, targets);
+
+        assertDsExclusivelyAssignedToTargets(targets, ds1.getId(), true, Status.RUNNING);
+
+        // Second assignment
+        final DistributionSet ds2 = testdataFactory.createDistributionSet("Multi-assign-2");
+        assignDistributionSet(ds2, targets);
+
+        assertDsExclusivelyAssignedToTargets(targets, ds2.getId(), true, Status.RUNNING);
+        assertDsExclusivelyAssignedToTargets(targets, ds1.getId(), true, Status.RUNNING);
+    }
+
+    private void assertDsExclusivelyAssignedToTargets(final List<Target> targets, final long dsId, final boolean active,
+            final Status status) {
+        final List<Action> assignment = actionRepository.findByDistributionSetId(PAGE, dsId).getContent();
+
+        assertThat(assignment).hasSize(10).as("Active = " + active).allMatch(action -> action.isActive() == active)
```

Comment:
```suggestion
        assertThat(assignment).hasSize(10).allMatch(action -> action.isActive() == active)
```
=> `.as("Active = " + active)` is not a description for `.hasSize(10)`